### ISSUE TITLE
Blazer queries causing GEOMG to temporarily error

### ISSUE THREAD

**Issue by karenmajewicz** [geobtaa/geomg#234]
New issue I noticed today:
- When I try to use bring up one of the Blazer queries, the application seems to time out. It will try to load the query, but eventually times out with a notification that an error has occurred. Afterwards, the whole application will show the Rails error message "We're sorry, but something went wrong" for a few minutes

- GEOMG will start working again after a few minutes. Does Blazer require too much memory to run or is there something else going on?

**Comment by ewlarson** [comment]
Query that will error: https://geomg.lib.umn.edu/blazer/queries/22-all-records-by-language

**Comment by ewlarson** [comment]
Per @mberkowski, this is NGINX timing out.

**Comment by mberkowski** [comment]
To get this running today, I set a 30min timeout for NGINX to wait for a response from Rails. This is strictly a slow query issue, with the database resources we have allocated on AWS. We only run a low powered DB because that's all we would typically need for a website backend. But reporting queries with aggregation and JSON attribute parsing are a different animal, maybe needing more cores or more RAM. I don't know much about how postgresql allocates resources JSON queries, but it is really hard to optimize for a reporting engine's queries because the whole idea is that you can build them arbitrarily.

For now, the query that was failing to run will succeed. But it will take TEN! MINUTES! so be patient with it. I am going to leave this issue open, and when we have the time we can experiment with heavier database resources on RDS. I don't know if it is possible to just scale it in place or if I'd need to deploy a new instance and migrate. RDS costs double with doubling database cores, so we can look for a balance.

**Comment by mberkowski** [comment]
Our database runs on a t3.micro instance, very tiny 1GB of RAM. https://aws.amazon.com/ec2/instance-types/t3/ The JSON reporting query probably eats that right up. I will experiment with a t3.medium (4GB) - would increase the cost by probably ~$100 a year if we paid a full year up front, or maybe twice that if we paid monthly.

**Comment by mberkowski** [comment]
This is working WAY BETTER now. I found that while the dev & prod DBs have exactly the same configuration and almost exactly the same dataset, the dev db was able to run this query in a few seconds while prod took ten minutes. Simple things like `SELECT DISTINCT type FROM kithe_models` took up to 10s to return 1 row.

I examined storage allocations on both and found that while neither was full, both were getting there. Extending the dev db's storage allocation was possible without downtime, and is followed by a storage optimization stage. I did the same on the prod DB and my guess is the storage optimization process (which would involve defragmentation, fsck operations) fixed up what was causing very slow reads on the prod host. It now runs this query in 11s. It's possible we had some filesystem corruption underlying the kithe_models that was killing reads on that table.

I'm surprised this didn't manifest in other ways though, like just really slow performance across GEOMG.
